src/form/Form.jsx

```diff
@@ -676,9 +676,14 @@ function Form({
 
       // Add a repeated row if the value went from unset to set
       // And this is the last field in a set of repeated fields
+      const isPreviousValueDefaultArray =
+        Array.isArray(previousValue) &&
+        Array.isArray(defaultValue) &&
+        previousValue.length === 0 &&
+        defaultValue.length === 0;
       if (
         index === repeatedRowCount - 1 &&
-        previousValue === defaultValue &&
+        (previousValue === defaultValue || isPreviousValueDefaultArray) &&
         value !== defaultValue
```

Comment:
similar to how you checked that the previous value was an empty array, you have to check here that the new value is not an empty array or value !== defaultvalue

Reply:
and below for repeated row removal, you also need a similar check that previousValue is not an empty array and the new value is the empty array

Reply:
this logic can probably all be decomposed